dali/operators/color/brightness_contrast.h

```diff
@@ -125,73 +54,87 @@ class BrightnessContrast : public Operator<Backend> {
-                 const ::dali::workspace_t<Backend> &ws) override {
-    const auto &input = ws.template InputRef<Backend>(0);
-    const auto &output = ws.template OutputRef<Backend>(0);
-    output_desc.resize(1);
-
-    TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
-        {
-          TypeInfo type;
-          type.SetType<OutputType>(output_type_);
-          output_desc[0] = {input.shape(), type};
-        }
-    ), DALI_FAIL("Unsupported image type"))  // NOLINT
-    return true;
+  void AcquireArguments(const ArgumentWorkspace &ws) {
+    OperatorBase::AcquireTensorArgument(brightness_, brightness_contrast::kBrightness, ws);
+    OperatorBase::AcquireTensorArgument(contrast_, brightness_contrast::kContrast, ws);
   }
 
+
+  USE_OPERATOR_MEMBERS();
+  std::vector<float> brightness_, contrast_;
+  DALIDataType output_type_;
+  kernels::KernelManager kernel_manager_;
+};
+
+
+class BrightnessContrastCpu : public BrightnessContrastOp<CPUBackend> {
+ public:
+  explicit BrightnessContrastCpu(const OpSpec &spec) : BrightnessContrastOp(spec) {}
+
   /*
    * So that compiler wouldn't complain, that
    * "overloaded virtual function `dali::Operator<dali::CPUBackend>::RunImpl` is only partially
    * overridden in class `dali::brightness_contrast::BrightnessContrast<dali::CPUBackend>`"
    */
-  using Operator<Backend>::RunImpl;
-
-
-  void RunImpl(Workspace<Backend> &ws) {
-    const auto &input = ws.template Input<Backend>(0);
-    auto &output = ws.template Output<Backend>(0);
-
-    TYPE_SWITCH(input.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
-        TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
-        {
-            using BrightnessContrastKernel =
-                    detail::BrightnessContrastKernel<Backend, OutputType, InputType, 3>;
-            auto tvin = view<const InputType, 3>(input);
-            auto tvout = view<OutputType, 3>(output);
-            kernel_manager_.Initialize<BrightnessContrastKernel>();
-            kernels::KernelContext ctx;
-            kernel_manager_.Setup<BrightnessContrastKernel>(ws.data_idx(),
-                    ctx, tvin, brightness_, contrast_);
-            kernel_manager_.Run<BrightnessContrastKernel>(ws.thread_idx(), ws.data_idx(),
-                    ctx, tvout, tvin, brightness_, contrast_);
-        }
-        ), DALI_FAIL("Unsupported output type"))  // NOLINT
-    ), DALI_FAIL("Unsupported input type"))  // NOLINT
+  using Operator<CPUBackend>::RunImpl;
+
+  ~BrightnessContrastCpu() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(BrightnessContrastCpu);
+
+ protected:
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<CPUBackend> &ws) override;
+
+  void RunImpl(workspace_t<CPUBackend> &ws) override;
+
+ private:
+  template <typename Kernel, typename InputType>
+  TensorListShape<> CallSetup(const TensorVector<CPUBackend> &input, int instance_idx) {
```

Comment:
Just a nitpick but maybe keep the similar order of arguments as kernel_manager_.Setup?